### PR TITLE
feat: filter contact search by appLabel [branch ch7811]

### PR DESCRIPTION
**Pull request by Karim94**
#### Relevant info and issue/PR links
https://app.clubhouse.io/peerio/story/7811/mobile-make-user-pickers-filterable-by-applabel
https://app.clubhouse.io/peerio/story/7501/mobile-filter-user-pickers-for-applabel-peerio-users-only

#### Testing instructions
We have 3 different contact lookup mechanisms, contact selector (share file, new dm, promote a user to admin, etc..), contact-add (contacts tab), and contact sync on sign up. These 3 mechanisms should be tested from both, Peerio and Medcryptor perspective. There's no need to test contact selector twice since its the same code. For example, if contact selector in share file works, then no need to test new dm or promote to admin since theyre the same.
Note: Peerio applies filtering to hide Medcryptor contacts. But Medcryptor currently applies no filtering, so all contacts should be visible.

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: app/components/contacts/contact-add.js [hunk 1]
```diff
@@ -120,10 +121,11 @@ export default class ContactAdd extends SafeComponent {
         this.waiting = true;
         this.toInvite = null;
         this.notFound = false;
-        const contact = contactStore.getContactAndSave(this.query);
+        const contact = contactStore.getContact(this.query);
         when(() => !contact.loading, () => {
             const { notFound, isLegacy, fullNameAndUsername: name } = contact;
-            if (notFound) {
+            const appLabelFilter = whiteLabel.filterAppLabel(contact);
```
Comment (seavan) [inline app/components/contacts/contact-add.js]: whiteLabel.filterAppLabel should set `contact.notFound = true` if applabel is medcryptor.
that way we don't need to check  for a separate appLabelFilter variable later on

File: app/components/contacts/contact-selector-universal.js [hunk 2]
```diff
@@ -151,8 +152,9 @@ export default class ContactSelectorUniversal extends SafeComponent {
         this.inProgress = true;
         when(() => !c.loading, () => {
             this.inProgress = false;
+            const appLabelFilter = whiteLabel.filterAppLabel(c);
```
Comment (seavan) [inline app/components/contacts/contact-selector-universal.js]: whiteLabel.filterAppLabel should set `contact.notFound = true` if applabel is medcryptor.
that way we don't need to check  for a separate appLabelFilter variable later on

**Comment by Karim94** [comment]
@seavan removed circular dependencies, but not sure if what I did was the best way to deal with it.

**Comment by flohdot** [comment]
this now has conflicts with dev/medcryptor-base... @seavan i'll let you handle it?

**Comment by seavan** [comment]
@flohdot yes, working on all three right now

**Comment by seavan** [comment]
@Karim94 merged. please test.

**Pull request merged**